### ISSUE TITLE
PostInstallScript and friends adds #!/bin/sh

### ISSUE THREAD

**Issue by TheAngryByrd** [qmfrederik/dotnet-packaging#85]
Currently the PostInstallScript (and others) adds `#!/bin/sh` to the script. 
 https://github.com/qmfrederik/dotnet-packaging/blob/63b290277b4c8b7d3f09f696c2307740f0a12e52/Packaging.Targets/Deb/DebPackageCreator.cs#L176

The way I've been using this is to have MSBuild read in a file from disk that I've specified as the postinstall script. I've already added my shebang to the file, but when I go to deploy it has double them now. I think this should be up the user to add their shebang.

**Comment by qmfrederik** [comment]
The reason `#!/bin/sh` get added is that other options (such as `CreateUser` or `InstallService`) dynamically construct a script, and we add the shebang at the very end to make the script valid.

Another way to handle this may be to store the user script "as is" in the package, and invoke it from the PreInstallScript.

**Comment by TheAngryByrd** [comment]
Or would it make sense to have a separate implementation for "inline" scripts and file paths?

**Comment by qmfrederik** [comment]
That may work as well. Would you like for that to override any "auto-generated" script (like for `CreateUser` or `InstallService`)?

**Comment by TheAngryByrd** [comment]
Yeah, if I'm providing my own scripts, I should bear the burden of setting up users/services correctly.
